src/types/events-types.ts

```diff
@@ -41,11 +42,36 @@ export type EventsGetReply = GenericMessageReply & {
   entries?: string[];
 };
 
+export type EventsSubscribeMessage = {
+  authorization?: AuthorizationModel;
+  descriptor: EventsSubscribeDescriptor;
+};
+
+export type EventsHandler = (message: GenericMessage) => void;
+
+export type EventsSubscription = {
+  id: string;
+  on: (handler: EventsHandler) => { off: () => void };
+  onError: (handler: (error: DwnError) => void) => void;
+  close: () => Promise<void>;
```

Comment:
Design question: What's the difference between `close()` and `off: () => void`? Is there a possibility to condense down to one?

Reply:
You can call the `.on()` method multiple times  with different handler functions on each, so the off that is returned will stop emitting to that specific handler. I figured it would be a better pattern to do that than add an `off()` method which must take the same exact handler.

`.close()` just closes the entire subscription no longer emitting to any handlers.

Reply:
Seems no longer applicable.